File: src/server/src/main/java/com/project/utilities/FindUtil.java
package com.project.utilities;

import java.util.Collection;
import java.util.Optional;
import java.util.function.Predicate;

public final class FindUtil {
    private FindUtil() {}

    public static <T> Optional<T> findByProperty(Collection<T> collection, Predicate<T> filter) {
        return collection.stream().filter(filter).findFirst();
    }
}

File: src/server/src/main/kotlin/com/project/utilities/FindUtil.kt
package com.project.utilities

import java.util.Optional
import java.util.function.Predicate

object FindUtil {
    @JvmStatic
    fun <T> findByProperty(collection: Collection<T?>? = emptyList(), filter: (T?) -> Boolean): T? {
        return collection?.firstOrNull(filter)
    }
}
